nailgun/entity_mixins.py

```diff
@@ -204,6 +205,77 @@ def _get_server_config():
     return config.ServerConfig.get()
 
 
+def _get_entity_id(field_name, attrs):
+    """Find the ID for a one to one relationship.
+
+    The server may return JSON data in the following forms for a
+    :class:`nailgun.entity_fields.OneToOneField`::
+
+        'user': None
+        'user': {'name': 'Alice Hayes', 'login': 'ahayes', 'id': 1}
+        'user_id': 1
+        'user_id': None
+
+    Search ``attrs`` for a one to one ``field_name`` and return its ID.
+
+    :param field_name: A string. The name of a field.
+    :param attrs: A dict. A JSON payload as returned from a server.
+    :returns: Either an entity ID or None.
+
+    """
+    field_name_id = field_name + '_id'
+    if field_name in attrs:
+        if attrs[field_name] is None:
+            return None
+        else:
+            return attrs[field_name]['id']
+    elif field_name_id in attrs:
+        return attrs[field_name_id]
+    else:
+        raise MissingValueError(
+            'Cannot find a value for the "{}" field. Searched for keys named '
+            '{}, but available keys are {}.'
+            .format(field_name, (field_name, field_name_id), attrs.keys())
+        )
+
+
+def _get_entity_ids(field_name, attrs):
```

Comment:
This method's docstring is worth reading. It provides examples of the different ways in which the server might hand back a response. This method is at the heart of this PR.